### DISCUSSION TITLE
Open source

### DISCUSSION THREAD

**Discussion by Aspie96** [stoatchat/discussions#137] [General]
(I am not a lawyer and this is not legal advice).

It is great that Revolt aims to be an open source tool.

Currently, some of the repositories needed to have a running copy of Revolt, however, do not have a license.

This means that all rights are reserved and that they are not open source.

I suggest adding a license ASAP, both to make it open source immediately, but more importantly so that contributions are also licensed under an OS license. If contributions had been received on those repos, this obviously requires the permission of those contributors.

Also, I notice you are using the Mutant emoji set.

Mutant is expressly not open source, has a proprietary license incompatible with commercial use and with the AGPL and relies on software (orxporter and forc) which is not under an open source license either.

I think an emoji set is not worth getting away from the open source spirit of the project and I suggest not using Mutant or at least making it not the default, in favour of an open source set.

**Comment by Aspie96** [comment]
Also MongoDB is proprietary now, but that is way harder to replace.

**Reply by sussycatgirl** [reply]
Wait what? Since when?

**Reply by Cyastis** [reply]
> Wait what? Since when?

In 2018 MongoDB's license changed from the AGPL to the SSPL, which is not a free software license according to the OSI and the DFSG.

Also by the way it seems Revolt's Delta just switched from MongoDB to Redis

**Reply by Zomatree** [reply]
> Also by the way it seems Revolt's Delta just switched from MongoDB to Redis

We switched to using redis for pubsub, all of the persistent data is still stored in mongodb

**Reply by Aspie96** [reply]
@janderedev I can confirm what @Cyastis said. You can see more info here: https://www.mongodb.com/licensing/server-side-public-license/faq And here: https://www.mongodb.com/community/licensing
Elastic also switched to the same license.

Note: I am not trying to say that Revolt should switch to another database or anything. It's just something I wanted to mention.

Of course replacing MongoDB would be great but I am fully aware that it would require much effort.

I think the priority should be adding the license file where it's missing: the fewer people have contributed at this, the easier it's to get all of them to agree (and it's the easiest thing to do).

**Reply by mokou** [reply]
Just because a few groups don't view the SSPL as an open-source license doesn't make it proprietary. It's practically the same as the AGPL unless you're Amazon/Google/whatever.

**Reply by Aspie96** [reply]
@Mokou you are free not to care, but the SSPL is absolutely not an open source license and is way different than the AGPL.
The SSPL is designed precisely so that the new requirement:
> .all programs that you use to make the Program or modified version available as a service, including, without limitation, management software, user interfaces, application program interfaces, automation software, monitoring software, backup software, storage software and hosting software, all such that a user could run an instance of the service using the Service Source Code you make available.

is effectively impossible to comply with.

The SSPL is not considered open source by OSI, it is not by Debian, it is not by Fedora.
Who exactly does consider it open surce, except maybe you?

MongoDB is source available but absolutely not open source.

**Reply by mokou** [reply]
@Aspie96 Good thing Revolt isn't a cloud hosting platform, or they might have to worry about that.

Debian, OSI, and RedHat don't get to define what is and is not "open source" because it's a vague term with various definitions. The OSI has been known to change their minds before as well, as was the case with the Unlicense. In 2012 they dismissed it as a crayon license and 8 years later it finally gained OSI approval.

My point is that you can call the SSPL open-source, source-available, or whatever you want, but it's not proprietary. MongoDB using that license has zero effect on the revoltchat/delta project as well.

**Reply by Aspie96** [reply]
> Debian, OSI, and RedHat don't get to define what is and is not "open source" because it's a vague term with various definitions.

Not really, though. It's a term with one definition (and interpretation of such definition) which is more recognized and widespread than pretty much any other definition (the definition that popularized the term).
It is recognized not only by the organization that popularized the term, but also, widely, by many companies, governments, government agencies and communities and this is strenghtened by such meaning being nearly identical to that of "free software".
I don't think the term is more vague than other widely used terms in computer science. In fact, it is one with a much wider consensus about its effective meaning than many other terms we use.
There are of course licenses for which different communities disagree about whether they are open source. This is not due to an ambiguity of the term, but rather to the fact that if pretty much anything is defined, there are going to be things (in this case licenses) at the very boundery of such group. That is not the situation of the SSPL license, however, which has a clause specifically meant to prevent running modified copies of the program and making them remotely accessible.

> My point is that you can call the SSPL open-source,

You can.
You can also call the moon a star, if you wish, but it doesn't make it one.
According to what meaning would the SSPL be open source?

> source-available,

MongoDB is source available, yes. The source code is there.

> or whatever you want, but it's not proprietary.

It's proprietary in the sense that it's non-free and not open source. I never denied the source code of MongoDB is available.

> MongoDB using that license has zero effect on the revoltchat/delta project as well.

The issue with using MongoDB isn't that any part of Revolt becomes proprietary as a result. Open source software can have proprietary dependencies. It also is not that there is anything illegal about it: if it were agains the AGPL adding an exception would be enough and I am not arguing it is against the SSPL either.

The issue is that it weakens one of the main advantages of Revolt. One advantage of Revolt (beyond just that of being self hosted) is being open source. If it has a proprietary dependency it becomes impossible to run an instance of Revolt without having to rely on proprietary software.

One can of course not care about software being open source. That's perfectly fine. But Revolt does care and is meant for an audience which does care. The open source nature of the project is mentioned right in the front page. And that claim isn't false, of course. But the advantage of it is greately weakened.

**Reply by erkinalp** [reply]
Query compatible MongoDB replacements exist, such as FerretDB: https://github.com/FerretDB/FerretDB

**Comment by mokou** [comment]
The mutant emoji set is under a Creative Commons license. It doesn't matter if it's compatible with the AGPL because an SVG doesn't "link" with any of the code, it's just served as is on it's own. Additionally, neither orxporter or forc code is found within this project, at least from what I can see, so it's license doesn't matter. The emoji SVG could have been created with photoshop or any other tool for all that it matters.
